MapleServer2/PacketHandlers/Game/BeautyHandler.cs

```diff
@@ -0,0 +1,611 @@
+            switch (shop.BeautyType)
+            {
+                case BeautyShopType.Hair:
+                    if (shop.BeautyCategory == BeautyCategory.Special)
+                    {
+                        voucherTag = "beauty_hair_special";
+                        break;
+                    }
+                    voucherTag = "beauty_hair";
+                    break;
+                case BeautyShopType.Face:
+                    voucherTag = "beauty_face";
+                    break;
+                case BeautyShopType.Makeup:
+                    voucherTag = "beauty_makeup";
+                    break;
+                case BeautyShopType.Skin:
+                    voucherTag = "beauty_skin";
+                    break;
+                case BeautyShopType.Dye:
+                    voucherTag = "beauty_itemcolor";
+                    break;
+                default:
+                    session.Send(NoticePacket.Notice("Unknown Beauty Shop", NoticeType.FastText));
+                    return;
+            }
+
+            Item voucher = session.Player.Inventory.Items.FirstOrDefault(x => x.Value.Tag == voucherTag).Value;
+            if (voucher == null)
+            {
+                session.Send(NoticePacket.Notice(SystemNotice.ItemNotFound, NoticeType.FastText));
+                return;
+            }
+
+            session.Send(BeautyPacket.UseVoucher(voucher.Id, 1));
+            InventoryController.Consume(session, voucher.Uid, 1);
+        }
+
+        private static void PayWithShopTokenCost(GameSession session, BeautyMetadata beautyShop)
+        {
+            int cost;
+            if (beautyShop.SpecialCost != 0)
+            {
+                cost = beautyShop.SpecialCost;
+            }
+            else
+            {
+                cost = beautyShop.TokenCost;
+            }
+
+            switch (beautyShop.TokenType)
+            {
+                case ShopCurrencyType.Meso:
+                    session.Player.Wallet.Meso.Modify(-(cost));
+                    break;
+                case ShopCurrencyType.ValorToken:
+                    session.Player.Wallet.ValorToken.Modify(-(cost));
+                    break;
+                case ShopCurrencyType.Treva:
+                    session.Player.Wallet.Treva.Modify(-(cost));
+                    break;
+                case ShopCurrencyType.Rue:
+                    session.Player.Wallet.Rue.Modify(-(cost));
+                    break;
+                case ShopCurrencyType.HaviFruit:
+                    session.Player.Wallet.HaviFruit.Modify(-(cost));
+                    break;
+                case ShopCurrencyType.Meret:
+                case ShopCurrencyType.GameMeret:
+                    session.Player.Wallet.Meret.Modify(-(cost));
+                    break;
+                case ShopCurrencyType.EventMeret:
+                    session.Player.Wallet.RemoveMerets(cost);
+                    break;
```

Comment:
If it's possible to use all types of merets just use `Wallet.RemoveMerets()`

```suggestion
                case ShopCurrencyType.Meret:
                case ShopCurrencyType.GameMeret:
                case ShopCurrencyType.EventMeret:
                    session.Player.Wallet.RemoveMerets(cost);
                    break;
```

Reply:
I believe this is how it works in game. If the currency cost is event merets, then yes, using `Wallet.RemoveMerets()` will work. However, if it's Meret or GameMeret, it cannot use EventMeret as an applicable currency.